### ISSUE TITLE
Run restorecon unconditionally

### ISSUE THREAD

**Issue by chewi** [BackSlasher/chef-selinuxpolicy#23]
I'm still getting to grips with SELinux but I'm having an issue where systemd is creating a UNIX socket with the wrong context. Possibly a systemd bug but I'm not sure. In any case, files can come and go and it is a little annoying that once an fcontext is added, restorecon is never run again unless you use the modify action. This is counter-intuitive and hard to script around. I think it should be run unconditionally. What do you think?

**Comment by BackSlasher** [comment]
I'm not 100% sure how SELinux works (yet :), but according to [this](http://serverfault.com/q/531431), creating a file (using `touch`) doesn't apply selinux fcontext rules but rather causes them to inherit from the parent (unless a transition is applied, which I don't know what it is).
Anyway, I considered this and I feel this is outside the cookbook's scope - the issue is not SELInux itself, but rather an annoying behavior of the creating program. If I were you, I'd solve it using an execute block, like this:
```ruby
execute 'fix-secontext-bla` do
  command 'restorecon /path/to/bla'
  not_if 'stat -c %C | grep REQUIRED_CONTEXT'
end
```
If you (or any other reader) see this pattern repeat in many locations, lmk and we can think about creating a resource for that / modifying the existing `fcontext`. If it's just a single instance, I think the execute block will suffice.

**Comment by chewi** [comment]
Yes, generally speaking a new file inherits from the parent. Unfortunately setting the context on the parent didn't work in this case. systemd seems to have some special handling around SELinux and sockets but I couldn't get it to do what I wanted and I'm not even sure that it can.

Adding an execute block is exactly what I did in the end. I could have limited it to the socket but after tripping up so many times following manual deletions, I decided to do it on the wider directory tree just to be sure. I haven't benchmarked it but restorecon seems pretty fast, even on a tree with many files.

I'm happy enough with this solution for now but thought I should at least mention the problem.
